### PR TITLE
TRD PulseHeight Task

### PR DESCRIPTION
**Pull request by maroozm**
PulseHeight Task is optimized as most of the code is now transferred to AliceO2. https://github.com/AliceO2Group/AliceO2/pull/11636

### CODE REVIEW

File: Modules/TRD/src/PulseHeightTrackMatch.cxx
Comment (martenole) [inline Modules/TRD/src/PulseHeightTrackMatch.cxx:55]: what does this plot actually tell us?

File: Modules/TRD/TRDQC.json
Comment (martenole) [inline Modules/TRD/TRDQC.json:130]: we don't need the pileup cut anymore in QC, this is a parameter in O2 now

File: Modules/TRD/src/PulseHeightTrackMatch.cxx
Comment (martenole) [inline Modules/TRD/src/PulseHeightTrackMatch.cxx:122]: could you make the type a configurable parameter? By default I would use everything as input that we have, but we should be able to select a certain type only

**Review by martenole — CHANGES_REQUESTED** [review]
Thanks, looks good. Just a few minor comments to address before merging

File: Modules/TRD/src/PulseHeightTrackMatch.cxx
Comment (maroozm) [reply Modules/TRD/src/PulseHeightTrackMatch.cxx:55]: i think, its the same mean of ADC per chamber.

File: Modules/TRD/TRDQC.json
Comment (maroozm) [reply Modules/TRD/TRDQC.json:130]: okay, will remove

File: Modules/TRD/src/PulseHeightTrackMatch.cxx
Comment (maroozm) [reply Modules/TRD/src/PulseHeightTrackMatch.cxx:122]: thanks. yes. i checked, it seems that value of the parameter is only 0 or 1. should keep 1 as default??

Comment (martenole) [reply Modules/TRD/src/PulseHeightTrackMatch.cxx:122]: the parameter can be anything from 0 to 3. By default everything should be accepted. But one should be able to select any combination of the parameters. So basically a bitmap would be good which has all 4 bits set by default

File: Modules/TRD/src/PulseHeightTrackMatch.cxx
Comment (martenole) [reply Modules/TRD/src/PulseHeightTrackMatch.cxx:55]: OK, and what can we learn from that plot? I have seen it once in a meeting I think and to me it is not clear what we should be able to see there

**Review by martenole — APPROVED** [review]
looks good

**Comment by knopers8** [comment]
It crashes in the fst:
```
reco_NOGPU.log-[45513:qc-task-TPC-Clusters]: [08:34:07][INFO] Using sampled data.
reco_NOGPU.log:[45535:qc-task-TRD-TRDPulseHeightTrackMatch]: [08:34:07][ERROR] Exception caught: InputRecord::get: no input with binding phdata found. Available inputs: timer-cycle, rawstats, triggers, digits, trigrectrk, tracks, tracklets 
reco_NOGPU.log-[45535:qc-task-TRD-TRDPulseHeightTrackMatch]: Executable is /sw/slc7_x86-64/QualityControl/1886-slc7_x86-64-local1/bin/o2-qc
reco_NOGPU.log-[45535:qc-task-TRD-TRDPulseHeightTrackMatch]:     /sw/slc7_x86-64/QualityControl/1886-slc7_x86-64-local1/lib/libO2QcTRD.so: decltype(auto) o2::framework::InputRecord::get<gsl::span<o2::trd::PHData, 18446744073709551615ul>, char const*>(char const*, int) const
--
reco_NOGPU.log-[45535:qc-task-TRD-TRDPulseHeightTrackMatch]:  Backtrace follow: 
reco_NOGPU.log-[45535:qc-task-TRD-TRDPulseHeightTrackMatch]: 
reco_NOGPU.log:[45535:qc-task-TRD-TRDPulseHeightTrackMatch]: terminate called after throwing an instance of 'o2::framework::RuntimeErrorRef'
reco_NOGPU.log:[45535:qc-task-TRD-TRDPulseHeightTrackMatch]: *** Program crashed (Segmentation fault, FPE, BUS, ABRT, KILL, Unhandled Exception, ...)
reco_NOGPU.log-[45535:qc-task-TRD-TRDPulseHeightTrackMatch]: Backtrace by DPL:
reco_NOGPU.log-[45535:qc-task-TRD-TRDPulseHeightTrackMatch]: ABRT
```

**Comment by maroozm** [comment]
> It crashes in the fst:
> 
> ```
> reco_NOGPU.log-[45513:qc-task-TPC-Clusters]: [08:34:07][INFO] Using sampled data.
> reco_NOGPU.log:[45535:qc-task-TRD-TRDPulseHeightTrackMatch]: [08:34:07][ERROR] Exception caught: InputRecord::get: no input with binding phdata found. Available inputs: timer-cycle, rawstats, triggers, digits, trigrectrk, tracks, tracklets 
> reco_NOGPU.log-[45535:qc-task-TRD-TRDPulseHeightTrackMatch]: Executable is /sw/slc7_x86-64/QualityControl/1886-slc7_x86-64-local1/bin/o2-qc
> reco_NOGPU.log-[45535:qc-task-TRD-TRDPulseHeightTrackMatch]:     /sw/slc7_x86-64/QualityControl/1886-slc7_x86-64-local1/lib/libO2QcTRD.so: decltype(auto) o2::framework::InputRecord::get<gsl::span<o2::trd::PHData, 18446744073709551615ul>, char const*>(char const*, int) const
> --
> reco_NOGPU.log-[45535:qc-task-TRD-TRDPulseHeightTrackMatch]:  Backtrace follow: 
> reco_NOGPU.log-[45535:qc-task-TRD-TRDPulseHeightTrackMatch]: 
> reco_NOGPU.log:[45535:qc-task-TRD-TRDPulseHeightTrackMatch]: terminate called after throwing an instance of 'o2::framework::RuntimeErrorRef'
> reco_NOGPU.log:[45535:qc-task-TRD-TRDPulseHeightTrackMatch]: *** Program crashed (Segmentation fault, FPE, BUS, ABRT, KILL, Unhandled Exception, ...)
> reco_NOGPU.log-[45535:qc-task-TRD-TRDPulseHeightTrackMatch]: Backtrace by DPL:
> reco_NOGPU.log-[45535:qc-task-TRD-TRDPulseHeightTrackMatch]: ABRT
> ```

Hi Piotr, the binding phdata works only if the `o2-trd-global-tracking` runs with option `--enable-ph`, and the json query: `TRD/PULSEHEIGHT`. everything works fine for me locally

**Comment by martenole** [comment]
Hi @maroozm 
but then we cannot merge it now. I'll add the `--enable-ph` option to the TRD global tracking by default, then the FST should work

**Comment by maroozm** [comment]
> Hi @maroozm but then we cannot merge it now. I'll add the `--enable-ph` option to the TRD global tracking by default, then the FST should work

hi, @martenole thanks. okay, i will change it to [WIP] until then

**Comment by martenole** [comment]
PR is here: https://github.com/AliceO2Group/AliceO2/pull/11672

**Comment by martenole** [comment]
Hi @maroozm can you remove the WIP again from the title? I have merged the PR in O2 so the PH data is available now

**Comment by maroozm** [comment]
> 3 CET on d233006c

HI @martenole , i edited the title. can i restart the tests?

**Comment by martenole** [comment]
They restart automatically. In fact you can already see that in the CI section:

> build/QualityControl/O2fst/o2 — Rechecking since Fri 12:23 CET on d233006c

**Comment by maroozm** [comment]
Hi, there seems to be an issue with o2fst. I do not understand why the o2fst is only accepting the following input binding `tracks, trigrectrk, digits, tracklets, triggers, rawstats ` while it should accept anything corresponding to the query in the json??

**Comment by martenole** [comment]
this does not make sense to me. Now the FST is green. Could you please retry and use your previous phValues instead of tracks as binding? If that then fails we have to understand it possibly by running the FST manually

**Comment by maroozm** [comment]
> this does not make sense to me. Now the FST is green. Could you please retry and use your previous phValues instead of tracks as binding? If that then fails we have to understand it possibly by running the FST manually

Hi @martenole, done. but i waited before changing it to tracks. there were more than 4 fsts, that gave the same error. but lets try again.

**Comment by maroozm** [comment]
Hi, it failed with `phValues`

**Comment by martenole** [comment]
Ok, I know what happens here. For the CI tests the QC json files are taken from O2DPG. Could you please change both https://github.com/AliceO2Group/O2DPG/blob/master/DATA/production/qc-sync/trditstpc.json and https://github.com/AliceO2Group/O2DPG/blob/master/DATA/production/qc-sync/trd.json
In those json files the `trdall` data sampling policy is defined. And that does knows only these sources:
```
"query": "digits:TRD/DIGITS/0;tracklets:TRD/TRACKLETS/0;triggers:TRD/TRKTRGRD/0;rawstats:TRD/RAWSTATS/0"
```

The TRDQC.json you changed here in your PR in QC is actually not used anywhere, except for in private and local tests.

**Comment by martenole** [comment]
Ok this should be good to go now that https://github.com/AliceO2Group/O2DPG/pull/1152 is merged. Not sure if the CI is restarted automatically. If not you might need to kick it with a dummy commit.

**Comment by maroozm** [comment]
why is this not merging?

**Comment by knopers8** [comment]
This is merging when someone with the write rights merges it, assuming the CI is green. It wasn't until now.

**Review by knopers8 — APPROVED** [review]

**Pull request merged**
